sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -494,15 +498,17 @@ class Analyzer(override val catalogManager: CatalogManager)
             case c @ Cast(ne: NamedExpression, _, _, _) => Alias(c, ne.name)()
             case e: ExtractValue =>
```

Comment:
We can handle `ExtractValue` better:
1. for field access like `some_expr.c`, use the field name as the alias
2. for map lookup with string literal like `map_col['key']`, use the key string as the alias
3. for others, invoke `optGenAliasFunc `